### PR TITLE
UI Scripts Tests

### LINKED ISSUE #129: Add tests for UI Code
Add various tests for the existing UI code in the project. [UI scripts folder](https://github.com/nicholas-maltbie/PropHunt/tree/master/Assets/Scripts/UI)

### PR DESCRIPTION
**Pull request by nicholas-maltbie**
# Description

Added tests for the UI Scripts.

Closes #129 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] ~~New and existing unit tests pass locally with my changes~~

(No testing framework setup yet so ignore the last two)

### CODE REVIEW

File: Assets/Scripts/UI/ConnectAction.cs [hunk 1]
```diff
@@ -38,25 +39,24 @@ public void ConnectToServer()
...
-                ushort.TryParse(serverPort.text, out networkPort);
-
-                // Verify parameters
-                // TODO: write verify code
+            // Parse parameters
+            bool validIP = IPAddress.TryParse(networkAddress, out parsedAddress);
+            bool validPort = ushort.TryParse(serverPort.text, out networkPort);
 
+            // Verify parameters
+            // TODO: write verify code
```
Comment (Oskar619) [inline Assets/Scripts/UI/ConnectAction.cs:47]: What would this verify code be?
Some sort of way of verifying the ip and port strings are well formed?
If so, I guess some googled regex patterns should do the trick.

**Review by Oskar619 — APPROVED** [review]
This looks great!

File: Assets/Scripts/UI/ConnectAction.cs (hunk 1 above)
Comment (nicholas-maltbie) [reply Assets/Scripts/UI/ConnectAction.cs:47]: Yeah, I'll make an issue for this. Right now there is just a try parse but it should be improved in the future to give user feedback through things like text, UI elements, progress meter, and everything else we need. 

Should also try to detect timeout when trying to connect or give feedback on why a request is rejected. 

Very simple state machine just need to scope it out and write it down.

Comment (nicholas-maltbie) [reply Assets/Scripts/UI/ConnectAction.cs:47]: https://github.com/nicholas-maltbie/PropHunt/issues/136 Issue created

**Pull request merged**